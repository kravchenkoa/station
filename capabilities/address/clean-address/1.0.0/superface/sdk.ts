import { createTypedClient } from "@superfaceai/one-sdk";

import { addressCleanAddress } from "./types/address/clean-address";

export { AddressCleanAddressProfile } from "./types/address/clean-address";
export const typeDefinitions = {
    ...addressCleanAddress
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
