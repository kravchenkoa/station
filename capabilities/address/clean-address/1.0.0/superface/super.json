{
  "profiles": {
    "address/clean-address": {
      "file": "./grid/profile.supr",
      "providers": {
        "smartystreets": {
          "file": "./grid/maps/smartystreets.suma"
        }
      }
    }
  },
  "providers": {
    "smartystreets": {
      "file": "../../../../../providers/smartystreets.json",
      "security": []
    }
  }
}