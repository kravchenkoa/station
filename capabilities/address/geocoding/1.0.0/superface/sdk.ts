import { createTypedClient } from "@superfaceai/one-sdk";

import { addressGeocoding } from "./types/address/geocoding";

export { AddressGeocodingProfile } from "./types/address/geocoding";
export const typeDefinitions = {
    ...addressGeocoding
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
