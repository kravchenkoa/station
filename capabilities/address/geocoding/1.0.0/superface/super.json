{
  "profiles": {
    "address/geocoding": {
      "file": "./grid/profile.supr",
      "providers": {
        "opencage": {
          "file": "./grid/maps/opencage.suma"
        },
        "google-apis": {
          "file": "./grid/maps/google-apis.suma"
        }
      }
    }
  },
  "providers": {
    "opencage": {
      "file": "../../../../../providers/opencage.json",
      "security": []
    },
    "google-apis": {
      "file": "../../../../../providers/google-apis.json",
      "security": []
    }
  }
}