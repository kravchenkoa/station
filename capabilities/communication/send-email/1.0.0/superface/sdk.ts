import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendEmail } from "./types/communication/send-email";

export { CommunicationSendEmailProfile } from "./types/communication/send-email";
export const typeDefinitions = {
    ...communicationSendEmail
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
