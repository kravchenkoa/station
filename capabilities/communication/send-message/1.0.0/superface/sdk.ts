import { createTypedClient } from "@superfaceai/one-sdk";

import { communicationSendMessage } from "./types/communication/send-message";

export { CommunicationSendMessageProfile } from "./types/communication/send-message";
export const typeDefinitions = {
    ...communicationSendMessage
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
