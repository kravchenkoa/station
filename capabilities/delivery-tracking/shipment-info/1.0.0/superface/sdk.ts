import { createTypedClient } from '@superfaceai/one-sdk';

import { deliveryTrackingShipmentInfo } from './types/delivery-tracking/shipment-info';

export { DeliveryTrackingShipmentInfoProfile } from "./types/delivery-tracking/shipment-info";
export const typeDefinitions = {
    ...deliveryTrackingShipmentInfo
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
