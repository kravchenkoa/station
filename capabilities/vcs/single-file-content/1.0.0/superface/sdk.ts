import { createTypedClient } from '@superfaceai/one-sdk';

import { vcsSingleFileContent } from './types/vcs/single-file-content';

export { VcsSingleFileContentProfile } from "./types/vcs/single-file-content";
export const typeDefinitions = {
    ...vcsSingleFileContent
};
export const SuperfaceClient = createTypedClient(typeDefinitions);
export type SuperfaceClient = InstanceType<typeof SuperfaceClient>;
