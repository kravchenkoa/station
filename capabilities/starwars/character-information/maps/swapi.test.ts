import * as fs from 'fs';
import { SuperfaceClient } from '../../../../superface/sdk';
// import nock from 'nock';
import { back } from 'nock';

back.fixtures = __dirname + '/nockFixtures';
back.setMode('record')

describe('starwars/character-information/swapi', () => {
  it('performs correctly', async () => {
    const client = new SuperfaceClient();
    const profile = await client.getProfile('starwars/character-information');
    const useCase = profile.useCases.RetrieveCharacterInformation;
    const provider = await client.getProvider('swapi');
    const inputs = {
      characterName: 'Luke Skywalker',
    };

    await testUseCase({ useCase, provider, inputs });
  });
});

// We would abstract this away in some function call
async function testUseCase({ useCase, provider, inputs }: any) {
  expect(useCase).not.toBeUndefined();
  expect(provider).not.toBeUndefined();

  let result: any;

  // We delete the existing fitures. There shoudl be a better way to do this.
  if (process.env.TEST === 'integration') {
    fs.unlinkSync(__dirname + '/nockFixtures/swapi-fixture.json')
  }

  // If the file doesn't exist, it creates one. If it does, it uses what exists in the file.
  const { nockDone } = await back('swapi-fixture.json');
  result = await useCase.perform(inputs, { provider });
  nockDone();

  // Instead of checking the output, we take a snapshot
  expect(result.unwrap()).toMatchSnapshot();
}
