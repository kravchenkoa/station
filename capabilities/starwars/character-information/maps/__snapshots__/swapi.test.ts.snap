// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starwars/character-information/swapi performs correctly 1`] = `
Object {
  "height": "172",
  "weight": "77",
  "yearOfBirth": "19BBY",
}
`;
